Add privacy consideration section (#19)
ietf-wg-acme/draft-bweeks-acme-device-attest: draft-ietf-acme-device-attest.md

The privacy analysis below addresses the two phases in which these identifiers appear: the attestation exchange between the client and ACME server during challenge validation, and the optional embedding of identifiers in the issued certificate.

## Identification and Correlation

The `permanent-identifier` type encodes a manufacturer assigned device identity, typically a serial number. The `hardware-module` type encodes the identity of the secure cryptoprocessor that generated the certificate key. In both cases, the identifier is globally unique within its assigner scope and unchanging for the lifetime of the device or hardware module.

From the perspective of {{!RFC6973}} Section 5.2.2, such identifiers enable direct identification of a device across protocol interactions, deployments, and time. Any entity that receives or observes these identifiers, including the ACME server, intermediary infrastructure, and any relying party that processes the issued certificate acquires an observable reference that can be used to track the device's certificate issuance history, renewal patterns, and operational context.

> When the same `permanent-identifier` or `hardware-module` value appears across multiple certificate requests (as it will in any recurring renewal workflow), it enables {{!RFC6973}} correlation: an observer with access to ACME server logs or certificate transparency logs can reconstruct the full lifecycle of a device's certificate activity. In enterprise environments where certificate issuance is logged centrally, this produces a persistent device audit trail regardless of whether the organization intends to maintain one.

Reviewer: ```suggestion
When the same `permanent-identifier` or `hardware-module` value appears across multiple certificate requests (as it will in any recurring renewal workflow), it enables {{!RFC6973}} correlation: an observer with access to ACME server logs can reconstruct the full lifecycle of a device's certificate activity. Similarly, when such identifiers are included in issued certificates, logging issued certificates in a central location (in certificate transparency logs, etc.) produces a persistent device audit trail regardless of whether the log operator intends to maintain one.
```

Implementers SHOULD assess whether the operational benefit of unchanging device identification outweighs this correlation exposure. In deployments where device anonymity or pseudonymity is a requirement, such as systems handling sensitive workloads on behalf of individuals, implementers SHOULD consider whether alternative validation mechanisms that do not bind the certificate to a permanent hardware identifier are more appropriate.

## Fingerprinting via Attestation Payloads

The `device-attest-01` challenge response carries a WebAuthn attestation object that may contain significantly more information than the identifier value alone. Depending on the attestation format, this payload may include device model, firmware version, bootloader state, hardware security level, and operating system version. Even when the resulting certificate is issued in a privacy-preserving form that omits the identifier from the subjectAltName extension (see Section 3.2 and Section 4.2), the attestation payload itself is transmitted to and evaluated by the ACME server during challenge validation.

> This constitutes a fingerprinting surface as defined in {{!RFC6973}} Section 3.2. The combination of a hardware serial number, hardware type OID, and firmware attestation attributes may uniquely identify not just the device model but the specific device unit, even in the absence of an explicit `permanent-identifier` value. Implementers operating ACME servers may consider applying data minimization principles to attestation payload handling by limiting only the attributes necessary to make the authorization decision should be evaluated, and the full attestation payload should not be retained beyond the duration of the challenge validation exchange unless there is a specific, documented operational requirement to do so.

Reviewer: ```suggestion
This constitutes a fingerprinting surface as defined in {{!RFC6973}} Section 3.2. The combination of a hardware serial number, hardware type OID, and firmware attestation attributes may uniquely identify not just the device model but the specific device unit, even in the absence of an explicit `permanent-identifier` value. Implementers operating ACME servers may consider applying data minimization principles to attestation payload handling by limiting only the attributes necessary to make the authorization decision should be evaluated, and the full attestation payload SHOULD NOT be retained beyond the duration of the challenge validation exchange unless there is a specific, documented operational requirement to do so.
```

- If the certificate is intended for use in certificate transparency logs, implementers MUST consider that embedding a `permanent-identifier` or `hardware-module` value will make that identifier permanently and publicly discoverable, indexed by issuance time, issuer, and subject. This constitutes an irreversible disclosure under {{!RFC6973}} Section 5.2.4 and should be avoided unless public discoverability of the device identifier is an explicit operational requirement.

## Stored Data and Account Binding

This document recommends the use of externalAccountBinding to pre-authenticate device requests to an enterprise ACME server. When an ACME account is persistently bound to a device identity, the server's account store contains a durable mapping between the cryptographic account credential and the physical device. Per {{!RFC6973}} Section 5.1.2, this stored association constitutes a target for compromise: an attacker who obtains the account store gains not only account credentials but a historical record of device-to-identity mappings across all certificate issuances.

> Implementers operating ACME servers should store account-to-device bindings using the minimum fidelity necessary for authorization decisions. Where the operational requirement is only to confirm that a given device is authorized to request certificates, it may be sufficient to store a hash or other one-way transformation of the device identifier rather than the identifier itself. Implementers should also define and enforce retention limits on historical account-to-certificate linkage records.

Reviewer: ```suggestion
Implementers operating ACME servers SHOULD store account-to-device bindings using the minimum fidelity necessary for authorization decisions. Where the operational requirement is only to confirm that a given device is authorized to request certificates, it may be sufficient to store a hash or other one-way transformation of the device identifier rather than the identifier itself. Implementers should also define and enforce retention limits on historical account-to-certificate linkage records.
```
